src/Endpoints/class-base-endpoint.php

```diff
@@ -156,4 +159,37 @@ public function register_endpoint(
 
 		register_rest_route( 'wp-parsely/v1', $endpoint, $rest_route_args );
 	}
+
+	/**
+	 * Registers the endpoint's WP REST route with arguments.
+	 *
+	 * @since 3.16.0
+	 *
+	 * @param string                      $endpoint The endpoint's route.
+	 * @param string                      $callback The callback function to call when the endpoint is hit.
+	 * @param array<string>               $methods The HTTP methods to allow for the endpoint.
+	 * @param array<WP_HTTP_Request_Args> $args The arguments for the endpoint.
+	 */
+	public function register_endpoint_with_args(
+		string $endpoint,
+		string $callback,
+		array $methods = array( 'GET' ),
+		array $args = array()
+	): void {
+		if ( ! apply_filters( 'wp_parsely_enable_' . convert_endpoint_to_filter_key( $endpoint ) . '_endpoint', true ) ) {
```

Comment:
Based on the feedback above, I'd make the same change here, for now.

```suggestion
		if ( ! apply_filters( 'wp_parsely_enable_' . convert_endpoint_to_filter_key( $endpoint ) . '_api_proxy', true ) ) {
```